python/cudf/cudf/core/column/column.py

```diff
@@ -1450,17 +1450,28 @@ def isin(self, values: Sequence) -> ColumnBase:
         result: Column
             Column of booleans indicating if each element is in values.
         """
-        try:
-            lhs, rhs = self._process_values_for_isin(values)
-            res = lhs._isin_earlystop(rhs)
-            if res is not None:
-                return res
-        except ValueError:
-            # pandas functionally returns all False when cleansing via
-            # typecasting fails
+        lhs, rhs = self._process_values_for_isin(values)
+        if lhs.dtype != rhs.dtype:
+            if lhs.null_count and rhs.null_count:
+                return lhs.isnull()
+            else:
+                return as_column(
+                    False, length=len(self), dtype=np.dtype(np.bool_)
```

Comment:
Do you know whether we have a test for `Series[int].isin(array[float])` where one of the floats compare equal?

```python
>>> pd.Series([1, 2]).isin([1.0, 2.5])
0     True
1    False
dtype: bool
```

I can't tell from the diff whether we'll return `[True, False]` or not.

Reply:
I think we have 1 combination in `test_isin_numeric`, but it's not entirely obvious if that is an intended test case for this scenario (`cudf.Series([0, 12, 14]).isin([1.0, 12.0, None, None, 120])`).

I'll add your test case just in case.